base/shell.jl

```diff
@@ -33,23 +33,25 @@ function shell_parse(str::AbstractString, interpolate::Bool=true;
     st = Iterators.Stateful(pairs(s))
 
     function update_arg(x)
-        if !isa(x,AbstractString) || !isempty(x)
-            push!(arg, x)
+        let arg = arg
```

Comment:
I wouldn't expect this to do anything --- `arg` has a declared type, plus this function already only reads the variable.

Reply:
And it doesn't help, it's still boxed. I meant to back that part out, will do it now. `s` and `i` are now unboxed.